src/mame/audio/leland.cpp

```diff
@@ -471,23 +462,15 @@ WRITE8_MEMBER( leland_80186_sound_device::leland_80186_control_w )
  *
  *************************************/
 
-void leland_80186_sound_device::command_lo_sync(void *ptr, int param)
-{
-	if (LOG_COMM) logerror("%s:Write sound command latch lo = %02X\n", machine().describe_context(), param);
-	m_sound_command = (m_sound_command & 0xff00) | param;
-}
-
-
 WRITE8_MEMBER( leland_80186_sound_device::leland_80186_command_lo_w )
 {
-	machine().scheduler().synchronize(timer_expired_delegate(FUNC(leland_80186_sound_device::command_lo_sync), this), data);
+	m_soundlatch->write(space, offset, data & 0xff, mem_mask & 0xff);
 }
 
 
 WRITE8_MEMBER( leland_80186_sound_device::leland_80186_command_hi_w )
 {
-	if (LOG_COMM) logerror("%s:Write sound command latch hi = %02X\n", machine().describe_context(), data);
-	m_sound_command = (m_sound_command & 0x00ff) | (data << 8);
+	m_soundlatch->write(space, offset, data << 8, mem_mask << 8);
```

Comment:
This won't actually work - the 16-bit latch device doesn't honour the mask.  You need to check that your changes actually build and work properly.